language/move-lang/src/naming/translate.rs

```diff
@@ -37,6 +37,7 @@ impl ResolvedType {
 struct Context {
     errors: Errors,
     current_module: Option<ModuleIdent>,
+    involved_modules: BTreeSet<ModuleIdent>,
```

Comment:
nit I'd probably remove this and use the other maps

Reply:
naming/translate.rs was actually the first module ever written for the source language. I started there and worked up to parsing and down through typing. As such, there are some patterns of code I am not particularly proud of. 

You can keep this if you really want to, but all of these different scoped information structs should really be combined at some point into a single `BTreeMap<ModuleIdent, ModuleInfo>`

Reply:
Yes, having a single `BTreeMap<ModuleIdent, ModuleInfo>`would be perfect. In this case, I could piggyback on one of `scoped_types`, `scoped_functions`, or `scoped_constants` and I can go for `scoped_functions` if that is OK?

Reply:
Updated in 25b3700